include/drivers/spi.h

```diff
@@ -142,6 +142,75 @@ struct spi_cs_control {
 	gpio_dt_flags_t		gpio_dt_flags;
 };
 
+/**
+ * @brief Declare the struct device for a @p gpio_dev field from
+ *        a devicetree node identifier
+ *
+ * The @p node_id parameter is a devicetree node identifier for a
+ * device on a SPI bus. If this SPI device has a chip-select GPIO
+ * controller defined in the devicetree, this macro declares the
+ * corresponding 'struct device'. Otherwise, it expands to nothing.
+ *
+ * @param node_id Devicetree node identifier for a SPI device
+ */
+#define SPI_MAYBE_DECLARE_CS_DEVICE_DT(node_id)				\
```

Comment:
Should we use `COND` (conditionally) instead of `MAYBE`?

Reply:
Or `IF_ANY`.
But indeed MAYBE doesn't fit well

Reply:
Why not just `SPI_DECLARE_CS_DEVICE_DT`?  It already documents that if there isn't anything to declare it doesn't do anything.

Reply:
+1 for `SPI_DECLARE_CS_DEVICE_DT`

Reply:
> Why not just `SPI_DECLARE_CS_DEVICE_DT`?

Agreed, done.